packages/styled-components/src/test/props.test.js

```diff
@@ -38,4 +41,18 @@ describe('props', () => {
     TestRenderer.create(<Comp fg="red" />);
     expectCSSMatches('.b { border-width:0; }');
   });
+
+  it('should forward the "as" prop if "innerAs" is used', () => {
+    const Comp = ({ as: Component = 'div', ...props }) => <Component {...props} />;
+
+    const Comp2 = styled(Comp)`
+      background: red;
+    `;
+
+    expect(TestRenderer.create(<Comp2 innerAs="button" />).toJSON()).toMatchInlineSnapshot(`
```

Comment:
Does this test really confirm that it fixes the bug. The issue was never that "as" didn't get applied, it's that styling was dropped from the originally `styled` component...

Reply:
I added an additional assertion to the test to make sure the className contains the correct style.